packages/core/src/configuration/Configuration.ts

```diff
@@ -0,0 +1,76 @@
+import { CredentialProvider, Credentials } from '../credentials-provider';
+import { container, DependencyContainer, injectable } from 'tsyringe';
+
+export interface ConfigurationOptions {
+  credentials?: Credentials;
+  cluster: string;
+  credentialProviders?: CredentialProvider[];
```

Comment:
Optional fields must be placed last)
```suggestion
  cluster: string;
  credentials?: Credentials;
  credentialProviders?: CredentialProvider[];
```